python/pyspark/sql/dataframe.py

```diff
@@ -506,10 +506,12 @@ def hint(self, name, *parameters):
         if not isinstance(name, str):
             raise TypeError("name should be provided as str, got {0}".format(type(name)))
 
+        allowed_types = (basestring, list, float, int)
```

Comment:
BTW, we had to add [`versionchanged` directive](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-versionchanged) in the docstring too.

Reply:
It doesn't really change behavior since it is a bug. Does this also apply for bugfixes? Let me know.